File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        if(n == 1 || n == 2 || n == 3)
            return n;
        
        int pre = 1, curr = 2, total;
        
        for(int i = 2; i < n; i++){
            
            total = pre + curr;
            pre = curr;
            curr = total;     
        }
        
        return total;
        
    }
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        if n == 2 or n == 1 or n == 3:
            return n
        
        ptr1, ptr2 = 1, 2
        
        for _ in range(2, n):
            
            stepsum = ptr1 + ptr2
            ptr1 = ptr2
            ptr2 = stepsum
            
        return stepsum